skbio/sequence/_sequence.py

```diff
@@ -473,12 +453,15 @@ def __getitem__(self, indexable):
                     indexable = np.asarray(indexable)
                 else:
                     seq = np.concatenate(
-                        list(self._slices_from_iter(self._bytes, indexable)))
-                    if self._has_quality():
-                        qual = np.concatenate(list(self._slices_from_iter(
-                            self.quality, indexable)))
-
-                    return self._to(sequence=seq, quality=qual)
+                        list(_slices_from_iter(self._bytes, indexable)))
+                    index = _as_slice_if_single_index(indexable)
+                    pos_md_slices = list(_slices_from_iter(
+                                         self.positional_metadata, index))
+                    positional_metadata = \
+                        pd.concat(pos_md_slices).reset_index(drop=True)
```

Comment:
add tests to make sure we don't mutate the user's dataframe/ndarray if they provided one as positional metadata